samcli/lib/package/packageable_resources.py

```diff
@@ -322,13 +330,22 @@ def delete(self, resource_id, resource_dict):
         """
         if resource_dict is None:
             return
+
+        s3_info = self.get_s3_info(resource_dict)
+        self.uploader.delete_artifact(remote_path=s3_info["Key"], is_key=True)
+
+    def get_s3_info(self, resource_dict):
```

Comment:
this class is to handle the resources that publish their own artifacts to ECR, so I am not able to understand the S3 part on this class

Reply:
So the classes ResourceWithS3UrlDict and ResourceZip upload their artifacts to S3 and so they have the methods get_s3_info which extracts the s3_bucket and key from the template file for theses resources.